### PR TITLE
Do not throw when trying to delete a  deleted appointment

### PR DESCRIPTION
**Pull request by yaron1m**
I see such errors in the logs

### CODE REVIEW

File: packages/functions/src/coordinator/DeleteAppointmentsHandler.ts [hunk 1]
```diff
@@ -17,9 +17,22 @@ export default async function (
...
+  ]);
 
+  if (appointments.length === 0) {
+    if (request.onlyRemoveDonor) {
+      throw new Error(
+        "Cannot remove donor from missing appointment - " +
+          request.appointmentId
+      );
+    }
+    functions.logger.debug("No appointment to delete", request.appointmentId);
```
Comment (evbambly) [inline packages/functions/src/coordinator/DeleteAppointmentsHandler.ts:31]: Why is this error silent?

**Review by evbambly — APPROVED** [review]

File: packages/functions/src/coordinator/DeleteAppointmentsHandler.test.ts [hunk 2]
```diff
@@ -90,17 +90,23 @@ test("User that does not have the right hospital throws exception", async () =>
...
     action,
-    `Appointment not found. Id ${APPOINTMENT_ID}`
+    `Cannot remove donor from missing appointment - ${APPOINTMENT_ID}`
   );
 });
 
+test("No such appointment fails silently", async () => {
+  await createUser(CoordinatorRole.SYSTEM_USER);
+
+  await callFunction(APPOINTMENT_ID, false, COORDINATOR_ID);
```
Comment (evbambly) [inline packages/functions/src/coordinator/DeleteAppointmentsHandler.test.ts:107]: Shouldn't there be an assertion?

Comment (yaron1m) [reply packages/functions/src/coordinator/DeleteAppointmentsHandler.test.ts:107]: I implicitly assert there's no exception throwns

File: packages/functions/src/coordinator/DeleteAppointmentsHandler.ts (hunk 1 above)
Comment (yaron1m) [reply packages/functions/src/coordinator/DeleteAppointmentsHandler.ts:31]: That's the target of this PR, to not throw in such cases (cause it is not actually an error)

**Pull request merged**